### ISSUE TITLE
Magento isn't saving attribute if up-sell products added with different attribute sets that doesn't include attribute for main product

### ISSUE THREAD

**Issue by archonkulis** [magento/magento2#16583]
### Preconditions
1. Magento 2.2.4

### Steps to reproduce
1. Add attribute set with some multiple select attribute
2. Add another attribute set and include an attribute that doesn't exist in the first attribute set
3. Create a product with the attribute set from #2
4. Create another product with the attribute set from #1
5. Try changing the value for the multiple select attribute for the product creatd in #3
6. Save product

### Expected result
1. Attribute successfully saves and catalog_product_entity_varchar is filled with the attribute value

### Actual result
1. Attribute isn't saving and catalog_product_entity_varchar table is empty for that attribute and product

### My observations
It seems that it's because when you save the product in admin, in the code it goes to `vendor/magento/module-catalog/Controller/Adminhtml/Product/Initialization/Helper.php::setProductLinks()` function. In that function, in `foreach ($linkTypes as $linkType => $readonly) {` it loads the linked products: `$linkProduct = $this->productRepository->getById($linkData['id']);`

And for all the linked products, in the code it ends up in the `vendor/magento/module-eav/Model/Config.php::getEntityAttributes()` function, but before that it in the function `vendor/magento/module-eav/Model/Entity/AttributeLoader.php::loadAllAttributes()` it unsets those attributes. Somehow, it unsets for the main product attributes `$resource->unsetAttributes();` and later it isn't loading the attribute for the main product.

### How I fixed the issue

Basically, in the function `vendor/magento/module-catalog/Controller/Adminhtml/Product/Initialization/Helper.php::setProductLinks()` it loads the product `$linkProduct = $this->productRepository->getById($linkData['id'])` only to get it's sku later on `->setLinkedProductSku($linkProduct->getSku())`

I changed `$linkProduct = $this->productRepository->getById($linkData['id'])` to:
```
 $linkProduct = $this->productCollectionFactory->create()
                        ->addAttributeToSelect('sku')
                        ->addAttributeToFilter('entity_id', $linkData['id'])
                        ->getFirstItem();
```

So that it doesn't load the up-sell products and now the attribute is saving correctly.

**Comment by magento-engcom-team** [comment]
Hi @archonkulis. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me {$VERSION} instance` 

where `{$VERSION}` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (2.2-develop +). 
For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation.

@archonkulis do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by archonkulis** [comment]
@magento-engcom-team yes

**Comment by engcom-backlog-nickolas** [comment]
@archonkulis, thank you for your report. Unfortunately I'm not able to reproduce this issue by given steps. Could you please provide more detailed steps to reproduce?

**Comment by engcom-backlog-nickolas** [comment]
@archonkulis, we are closing this issue due to inactivity. If you'd like to update it, please reopen the issue.
